packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -5,332 +5,483 @@ const { Promise } = Cypress
+          expect(cookies).to.have.length(4)
+
+          const sortedCookies = Cypress._.sortBy(cookies, 'name')
+
+          expect(sortedCookies[0].name).to.equal('key3')
+          expect(sortedCookies[0].domain).to.match(/\.?www\.barbaz\.com/)
+          expect(sortedCookies[1].name).to.equal('key4')
+          expect(sortedCookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+          expect(sortedCookies[2].name).to.equal('key5')
+          expect(sortedCookies[2].domain).to.match(/\.?barbaz\.com/)
+          expect(sortedCookies[3].name).to.equal('key6')
+          expect(sortedCookies[3].domain).to.match(/\.?barbaz\.com/)
         })
       })
     })
+  })
 
-    it('returns cookies for all domains when domain is null', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('foo', 'bar')
+  context('#getCookie', () => {
+    const setCookies = () => {
+      cy.log('set cookies')
+      cy.setCookie('key', 'www.barbaz.com value', { domain: 'www.barbaz.com', log: false })
+      cy.setCookie('key', 'barbaz.com value', { domain: 'barbaz.com', log: false })
+      cy.setCookie('key', 'www.foobar.com value', { domain: 'www.foobar.com', log: false })
+      cy.setCookie('key', 'foobar.com value', { domain: 'foobar.com', log: false })
+    }
+
+    it('returns the cookie from the domain matching the AUT by default when AUT is an apex domain', () => {
+      cy.visit('http://barbaz.com:3500/fixtures/generic.html')
+      setCookies()
 
-      cy.getCookies({ domain: null }).should('have.length', 2)
+      cy.getCookie('key').then((cookie) => {
+        expect(cookie.value).to.equal('barbaz.com value')
+        expect(cookie.domain).to.match(/\.?barbaz\.com/)
+      })
 
-      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+      // webkit does not support cy.origin()
+      if (isWebkit) return
 
-      cy.origin('http://www.foobar.com:3500', () => {
-        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
+      cy.origin('http://foobar.com:3500', () => {
+        cy.visit('http://foobar.com:3500/fixtures/generic.html')
 
-        cy.getCookies({ domain: null }).should('have.length', 2)
+        cy.getCookie('key').then((cookie) => {
+          expect(cookie.value).to.equal('foobar.com value')
+          expect(cookie.domain).to.match(/\.?foobar\.com/)
+        })
       })
     })
-  })
 
-  context('#getCookie', () => {
-    it('returns the cookie from the domain matching the AUT by default', () => {
+    it('can return the cookie from the subdomain matching the AUT by default when AUT is a subdomain', () => {
       cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('foo', 'bar')
+      cy.setCookie('key', 'www.barbaz.com value', { domain: 'www.barbaz.com', log: false })
```

Comment:
what happens here if `setCookies()` is called? Does getCookie return the most specific cookie or is it just the first one it finds that matches?

Reply:
It returns the first one that matches, which is existing behavior.